### DISCUSSION TITLE
Задачи

### DISCUSSION THREAD

**Discussion by picolino** [habralab/garnet-team#134] [Идеи]
Гарнет - это продукт, концепция которого крутится вокруг социального и профессионального взаимодействия.

Мы хотим сделать сообщества результативными на практике, а это значит, что нам необходимо дать пользователям инструменты для реализации их проектов. Одним из главных инструментов такого характера является функционал задач.

Задача - это новая сущность, которая привязана к проекту и состоит из следующих свойств:

#### Наименование задачи
Текст + эмодзи.

#### Описание задачи
Поддержка полноценного markdown
Плюс есть возможность указать ссылку на другую задачу с использованием нотации `#xxx`

#### Ответственный
По умолчанию - это постановщик задачи (пользователь, который создает задачу). 
В поле можно выбрать любого участника проекта в качестве ответственного.

#### Команды-исполнители (0..n)
При создании задачи есть возможность указать соответствующие команды-исполнители.
Пока не выбрана команда-исполнитель выбор пользователей-исполнителей недоступен.

_Команда исполнитель нужна чтобы постановщик задачи не занимался распределением задач внутри конкретных команд, работающих над проектом_

#### Пользователи-исполнители (0...n)
Непосредственно исполнители задачи. Пользователи, которые будут по-факту выполнять поставленную задачу.

#### Теги
Теги навыков, которые относятся к задаче

#### Лейблы
Для удобства фильтрации (по аналогии с GitHub)

Задача жестко привязана к проекту.
Задачу в проект может завести любой участник проекта.
У владельца проекта повышенные привилегии в отношении всех задач проекта

**Comment by Demorey** [comment]
> Ответственный
> По умолчанию - это постановщик задачи (пользователь, который создает задачу).
> В поле можно выбрать любого участника проекта в качестве ответственного.

Предлагаю оставить просто Создателя, чтобы его нельзя было изменить, так как будут например ситуации когда пользователь который создал задачу покинул проект и возможно это будет определяющим фактором оставлять задачу в работе или нет. А ответственный это по факту и будут исполнители задачи которых смогут менять.

**Reply by picolino** [reply]
Нет, создатель задачи будет в аудитной информации
А ответственный по задаче - это отдельная бизнес-роль процесса исполнения задачи, от нее отказываться не нужно
